src/Resource/Processor/CpuCoresCountProvider.php

```diff
@@ -65,24 +66,30 @@ public static function provide(): int
             return 1;
         }
 
-        // for Linux
-        $hasNproc = trim(@shell_exec('command -v nproc'));
+        try {
+            // for Linux
+            $hasNproc = trim(@shell_exec('command -v nproc'));
 
-        if ($hasNproc !== '') {
-            $nproc = trim(shell_exec('nproc'));
-            $cpuCount = filter_var($nproc, FILTER_VALIDATE_INT);
+            if ($hasNproc !== '') {
+                $nproc = trim(shell_exec('nproc'));
+                $cpuCount = filter_var($nproc, FILTER_VALIDATE_INT);
 
-            if (is_int($cpuCount)) {
-                return $cpuCount;
+                if (is_int($cpuCount)) {
+                    return $cpuCount;
+                }
             }
+        } catch (ExecException) {
         }
 
-        // for MacOS
-        $ncpu = trim(shell_exec('sysctl -n hw.ncpu'));
-        $cpuCount = filter_var($ncpu, FILTER_VALIDATE_INT);
+        try {
+            // for MacOS
+            $ncpu = trim(shell_exec('sysctl -n hw.ncpu'));
+            $cpuCount = filter_var($ncpu, FILTER_VALIDATE_INT);
 
-        if (is_int($cpuCount)) {
-            return $cpuCount;
+            if (is_int($cpuCount)) {
+                return $cpuCount;
+            }
+        } catch (ExecException) {
```

Comment:
That being said I'd personally find unexpected that you ask several threads and end up with only one... but that's out of the scope of this PR

Reply:
Yes, maybe we could add a notice/warning if all commands failed? WDYT?

Reply:
👍 (would do so in a different PR though)

Reply:
When/if this PR is merged I'll submit a new one where we can discuss this. 😉